src/components/totals/FundsRaised/index.js

```diff
@@ -58,26 +61,39 @@ module.exports = React.createClass({
     return campaignUids;
   },
 
-  loadCampaigns: function() {
+  loadTotals: function() {
     this.setState({ isLoading: true });
-    campaigns.findByUids(this.setUids(), this.onSuccess);
+
+    if (this.props.pageId && (this.props.campaignUid || this.props.campaignUids.length>0)) {
```

Comment:
This is a pretty chunky block of code. I wonder if we might be able to break this down somehow.